docs(@clayui/form): Fix headings (#2589)
liferay/clay: next.clayui.com/content/docs/components/forms/input.mdx

---
title: "Input"
description: "This section demonstrates the different text input types."
packageNpm: "@clayui/form"
packageVersion: "3.0.0"
packageStatus: "Stable"
sibling: "docs/components/forms/css-text-input.html"
---

import {
   FormInput,
   FormInputGroupConnected,
   FormInputGroupMixed,
   FormInputGroupSeparated,
   FormInputTextArea,
} from '../../../../src/components/clay/Input';

<div class="nav-toc-absolute">
<div class="nav-toc">

- [Basic Usage](#basic-usage)
- [Groups](#groups)
  - [Separated](#separated)
  - [Connected](#connected)
  - [Mixed](#mixed)
  - [Stacked](#stacked)
- [API](#api)

</div>
</div>

<div class="clay-site-alert alert alert-warning">
   Take a look at our <a href="https://storybook.clayui.com/?path=/story/components-clayinput--separated">Storybook</a> for more information about this component.
</div>

<div class="clay-site-alert alert alert-info">
	Check the <a href="https://liferay.design/lexicon">Lexicon</a> <a href="https://liferay.design/lexicon/core-components/forms/text-input/">Input and Input Groups Pattern</a> for a more in-depth look at the motivations and proper usage of this component.
</div>

> `ClayInput` is an aggregate component of the `@clayui/form` package, consisting of some low-level utilities that provides the ability to create Inputs and Input Groups.

Reviewer: ```suggestion
`ClayInput` is exported from the `@clayui/form` package, consisting of some low-level utilities that provides the ability to create Inputs and Input Groups.
```

> `<ClayInput />` enables you to make compositions to achieve your expected results, as this component is distributed only in a low level component, this increases flexibility and you can create many possibilities but it comes with a price tag... you may want to get close to the Lexicon specifications and will have to compose with some of the other components to get there, let's detail it further below.

Reviewer: I'm not sure this block is very helpful here. I don't think we need to over explain the low-level aspect of this and we could probably simply just put that ClayInput is the lexicon specification of input based in react.

Author: I think we can just mention about our composition pattern and link to our 
Composition docs that Matu recently wrote. What do you think?

Author: Like:

`ClayInput` is exported from the `@clayui/form` package, consisting of some <a href="https://next.clayui.com/docs/foundations/composing.html">low-level</a> utilities that provides the ability to create Inputs and Input Groups.

Author: Composition Docs: https://github.com/liferay/clay/pull/2585

Reviewer: Yeah that example would be great and just having a link back to the other docs.

  - [Stacked](#stacked)
- [API](#api)

</div>
</div>

<div class="clay-site-alert alert alert-warning">
   Take a look at our <a href="https://storybook.clayui.com/?path=/story/components-clayinput--separated">Storybook</a> for more information about this component.
</div>

<div class="clay-site-alert alert alert-info">
	Check the <a href="https://liferay.design/lexicon">Lexicon</a> <a href="https://liferay.design/lexicon/core-components/forms/text-input/">Input and Input Groups Pattern</a> for a more in-depth look at the motivations and proper usage of this component.
</div>

<div class="clay-site-alert alert alert-warning">
	<strong class="lead">Warning</strong>
	Before you get started with this doc, read about <a href="/foundations/composing.html" target="_blank">Clay's fundamentals of composition</a> to move on. The information below is assuming you have read about <a href="/foundations/composing.html" target="_blank">Clay's compositional philosophy</a> and learned about the terms.
</div>

`ClayInput` is exported from the `@clayui/form` package, consisting of some low-level utilities that provides the ability to create Inputs and Input Groups.

### Basic Usage

<FormInput />

> ### Using another elements as an input

Reviewer 2: Add this to the ToC as well.
